### PR TITLE
Add 5 more cops for LWRP deprecations in Chef 13

### PR DESCRIPTION
**Pull request by tas50**
Add 5 more cops for LWRP deprecations in Chef 13

ChefDeprecations/ResourceOverridesProvidesMethod
ChefDeprecations/ResourceUsesDslNameMethod
ChefDeprecations/ResourceUsesUpdatedMethod
ChefDeprecations/NamePropertyWithDefaultValue
ChefDeprecations/ResourceUsesProviderBaseMethod:

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,3 +1,47 @@
+## Cookstyle 5.7
+
+### 5 New Chef Cops
+
+#### ChefDeprecations/ResourceOverridesProvidesMethod
+
+The `ChefDeprecations/ResourceOverridesProvidesMethod` cop detects overriding the `provides?` method in a resource's provider without also calling `provides :SOME_PROVIDER_NAME`. This will cause failures in Chef Infra Client 13 and later.
```
Comment (lamont-granquist) [inline RELEASE_NOTES.md]: `provides?` should just get dropped though in favor of the provides dsl line.

File: docs/cops_chefdeprecations.md [hunk 2]
```diff
@@ -558,6 +583,116 @@ Name | Default value | Configurable values
...
+def provides?
+ true
+end
+
+# good
+provides :SOME_PROVIDER_NAME
+
+def provides?
+ true
+end
```
Comment (lamont-granquist) [inline docs/cops_chefdeprecations.md]: still bad.

Comment (tas50) [reply docs/cops_chefdeprecations.md]: Updated

File: RELEASE_NOTES.md (hunk 1 above)
Comment (tas50) [reply RELEASE_NOTES.md]: Updated

**Pull request merged**